wicker/core/storage.py

```diff
@@ -463,92 +487,141 @@ class S3PathFactory(WickerPathFactory):
+    def get_dataset_schema_path(
+        self, dataset_id: DatasetID, s3_prefix: bool = True, cut_prefix_override: Optional[str] = None
+    ) -> str:
         """Get path to the dataset schema.
 
         Args:
             dataset_id (DatasetID): ID of the dataset.
             s3_prefix (bool, optional): Whether to keep the s3 prefix or not. Defaults to True.
-
+            cut_prefix_override (str, optional): whether to use an arbitrary prefix to cut off path. Defaults to None.
         Returns:
             str: Path to dataset schema.
         """
-        prefix_to_trim = "s3://" if not s3_prefix else None
+        prefix_to_trim = self.__determine_cut_prefix(s3_prefix, cut_prefix_override)
         return self._get_dataset_schema_path(dataset_id=dataset_id, prefix_to_trim=prefix_to_trim)
 
-    def get_column_concatenated_bytes_files_path(self, s3_prefix: bool = True, dataset_name: str = None) -> str:
+    def get_column_concatenated_bytes_files_path(
+        self, s3_prefix: bool = True, dataset_name: str = None, cut_prefix_override: Optional[str] = None
+    ) -> str:
         """Gets the path to the root of all column_concatenated_bytes files
 
         :param s3_prefix: whether to return the s3:// prefix, defaults to True
         :param dataset_name: if self.store_concatenated_bytes_files_in_dataset is True,
             it requires dataset name, defaults to None
+        :param cut_prefix_override (str, optional): whether to use an arbitrary prefix to cut off path.
+            Defaults to None.
         :return: path to the column_concatenated_bytes file with the file_id
         """
-        prefix_to_trim = "s3://" if not s3_prefix else None
+        prefix_to_trim = self.__determine_cut_prefix(s3_prefix, cut_prefix_override)
         return self._get_column_concatenated_bytes_files_path(dataset_name=dataset_name, prefix_to_trim=prefix_to_trim)
 
-    def get_column_concatenated_bytes_s3path_from_uuid(self, file_uuid: bytes, dataset_name: str = None) -> str:
+    def get_column_concatenated_bytes_s3path_from_uuid(
+        self,
+        file_uuid: bytes,
+        dataset_name: str = None,
+        s3_prefix: bool = True,
+        cut_prefix_override: Optional[str] = None,
+    ) -> str:
         """Public gettr for column concat bytes with uuid
 
         Args:
             file_uuid (bytes): uuid of the file
             dataset_name (str, optional): Name of the dataset to gather. Defaults to None.
+            s3 (bool, optional): whether to return the s3:// prefix, defaults to True.
+            cut_prefix_override (str, optional): whether to use an arbitrary prefix to cut off path. Defaults to None.
 
         Returns:
             str: _description_
         """
-        columns_root_path = self.get_column_concatenated_bytes_files_path(dataset_name=dataset_name)
+        columns_root_path = self.get_column_concatenated_bytes_files_path(
+            dataset_name=dataset_name, s3_prefix=s3_prefix
```

Comment:
@pickles-bread-and-butter don't forget to pass `cut_prefix_override` to `self.get_column_concatenated_bytes_files_path`

Reply:
Thanks for pointing it out, done